### PR TITLE
Finalising Subscriptions types

### PR DESCRIPTION
**Pull request by darrellwarde**

### CODE REVIEW

File: packages/graphql/src/schema/subscriptions/generate-subscription-where-type.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        ...node.interfaceFields,
+        ...node.objectFields,
+        ...node.unionFields,
+        ...node.temporalFields,
+        ...node.pointFields,
+        ...node.computedFields,
+    ]);
+
+    return schemaComposer.createInputTC({
+        name: `${node.name}SubscriptionWhere`,
```
Comment (darrellwarde) [inline packages/graphql/src/schema/subscriptions/generate-subscription-where-type.ts:39]: I really don't like this type name...

File: packages/graphql/src/schema/subscriptions/generate-subscription-types.ts [hunk 2]
```diff
@@ -31,47 +34,77 @@ export function generateSubscriptionTypes({
...
 
         const nodeCreatedEvent = schemaComposer.createObjectTC({
             name: `${node.name}CreatedEvent`,
             fields: {
-                [subscribeOperation.created]: composeNode,
+                event: {
+                    type: eventTypeEnum.NonNull,
+                    resolve: () => EventType.getValue("CREATE"),
+                },
+                [`created${node.name}`]: {
```
Comment (angrykoala) [inline packages/graphql/src/schema/subscriptions/generate-subscription-types.ts:51]: :thinking: if we use node.name here we will have problems with underscores right?

Maybe we should expose `node.singular` or something like that to provide a standard name here

File: packages/graphql/src/schema/subscriptions/generate-subscription-types.ts [hunk 3]
```diff
@@ -31,47 +34,77 @@ export function generateSubscriptionTypes({
...
         const nodeUpdatedEvent = schemaComposer.createObjectTC({
             name: `${node.name}UpdatedEvent`,
             fields: {
-                [subscribeOperation.updated]: composeNode,
+                event: {
+                    type: eventTypeEnum.NonNull,
+                    resolve: () => EventType.getValue("UPDATE"),
+                },
+                previousState: {
+                    type: eventPayload.NonNull,
```
Comment (angrykoala) [inline packages/graphql/src/schema/subscriptions/generate-subscription-types.ts:66]: question: Will this work fine with nodes only having relationships?

File: packages/graphql/src/schema/subscriptions/generate-subscription-where-type.ts (hunk 1 above)
Comment (angrykoala) [reply packages/graphql/src/schema/subscriptions/generate-subscription-where-type.ts:34]: Maybe with some ketchup and mustard?

**Review by angrykoala — CHANGES_REQUESTED** [review]

**Review by angrykoala — DISMISSED** [review]

File: packages/graphql/src/schema/subscriptions/generate-subscription-types.ts (hunk 3 above)
Comment (darrellwarde) [reply packages/graphql/src/schema/subscriptions/generate-subscription-types.ts]: Following our conversation this morning, I believe the right approach is to not add the payload types if they have no actual value. This is now implemented.

**Pull request merged**